app/singleton.server.ts

```diff
@@ -0,0 +1,13 @@
+// Since the dev server re-requires the bundle, do some shenanigans to make
+// certain things persist across that 😆
+// Borrowed/modified from https://github.com/jenseng/abuse-the-platform/blob/2993a7e846c95ace693ce61626fa072174c8d9c7/app/utils/singleton.ts
+
+export const singleton = <Value>(
+  name: string,
+  valueFactory: () => Value
+): Value => {
+  const yolo = global as any;
+  yolo.__singletons ??= {};
+  yolo.__singletons[name] ??= valueFactory();
+  return yolo.__singletons[name];
+};
```

Comment:
We already mention this pattern in the docs and I think we should try to make the code as consistent with that as possible (https://remix.run/docs/en/main/other-api/dev-v2#keep-in-memory-data-and-connections-across-rebuilds)

Reply:
I just took over whatever @jenseng had in his code + adapted it to accept a function (just like Kent did in his personal website and the epic stack)